aten/src/ATen/native/Pad.cpp

```diff
@@ -0,0 +1,271 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/Resize.h>
+#include <ATen/TensorIndexing.h>
+
+namespace at { namespace native {
+
+// Restrides a pad specifier argument to size(ndim, 2). Using a standard shape
+// lets us simplify the logic required to use pad specifier arguments.
+// This function also checks that the pad specifier has a legal format.
+//
+// TODO: Consider creating a PadSpecifier class, which would wrap the output of
+// this function. Helper functions could take the PadSpecifier instead of
+// a Tensor, so that we never accidentally give them the unprocessed pad
+// specifier tensor.
+Tensor _expand_pad_specifier(const Tensor& pad_spec, c10::string_view arg_name, int64_t ndim) {
+  auto pad_spec_ndim = pad_spec.dim();
+  auto pad_spec_numel = pad_spec.numel();
+
+  // TODO: This logic can probably be simplified
+  if (pad_spec_numel == 1) {
+    // size [], [1], [2] cases
+    TORCH_CHECK(
+      (pad_spec_ndim >= 0) && (pad_spec_ndim <= 2),
+      "Expected ", arg_name, ".size() to be either ",
+      "[], [1], [2], [1, 1], [1, 2], [self.dim(), 1], or [self.dim(), 2], but got ",
```

Comment:
"input.dim()" in the user-facing error message because that's the name in the docs